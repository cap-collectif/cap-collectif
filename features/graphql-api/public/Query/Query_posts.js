/* eslint-env jest */

const query = /* GraphQL */ `
  query PostsQuery($orderBy: PostOrder, $theme: ID, $project: ID, $first: Int, $after: String) {
    posts(orderBy: $orderBy, theme: $theme, project: $project, first: $first, after: $after) {
      totalCount
      edges {
        node {
          id
          title
          publishedAt
          themes {
            id
            title
          }
          projects {
            id
            title
          }
          frTitle: title(locale: FR_FR)
          enTitle: title(locale: EN_GB)
          translations {
            title
            body
            abstract
            locale
          }
        }
      }
    }
  }
`;

describe('Internal|Query.posts array', () => {
  it('fetches the posts with their translations', async () => {
    await expect(graphql(query, null, 'internal')).resolves.toMatchSnapshot();
  });

  it('fetches the posts filtered by theme with their translations', async () => {
    await expect(
      graphql(
        query,
        {
          theme: 'theme1',
        },
        'internal',
      ),
    ).resolves.toMatchSnapshot();
  });

  it('fetches the posts filtered by project with their translations', async () => {
    await expect(
      graphql(
        query,
        {
          project: global.toGlobalId('project', 'project1'),
        },
        'internal',
      ),
    ).resolves.toMatchSnapshot();
  });

  it('fetches the posts filtered by theme and project with their translations', async () => {
    await expect(
      graphql(
        query,
        {
          theme: 'theme1',
          project: global.toGlobalId('project', 'ProjectAccessibleForMeOnlyByAdmin'),
        },
        'internal',
      ),
    ).resolves.toMatchSnapshot();
  });

  it('fetches the posts ordered by published date ascending with their translations', async () => {
    await expect(
      graphql(
        query,
        {
          orderBy: { field: 'PUBLISHED_AT', direction: 'ASC' },
        },
        'internal',
      ),
    ).resolves.toMatchSnapshot();
  });

  it('fetches the posts ordered by published date descending with their translations', async () => {
    await expect(
      graphql(
        query,
        {
          orderBy: { field: 'PUBLISHED_AT', direction: 'DESC' },
        },
        'internal',
      ),
    ).resolves.toMatchSnapshot();
  });

  it('fetches the first 2 posts with their translations', async () => {
    await expect(
      graphql(
        query,
        {
          first: 2,
        },
        'internal',
      ),
    ).resolves.toMatchSnapshot();
  });

  it('fetches the first 3 posts after one specific post with their translations', async () => {
    await expect(
      graphql(
        query,
        {
          first: 3,
          after: 'YXJyYXljb25uZWN0aW9uOjA='
        },
        'internal',
      ),
    ).resolves.toMatchSnapshot();
  });
});
