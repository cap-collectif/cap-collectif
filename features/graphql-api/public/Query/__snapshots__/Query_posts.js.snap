// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Internal|Query.posts array fetches the posts with their translations 1`] = `
Object {
  "posts": Array [
    Object {
      "enTitle": "Private project post for admin@test.com",
      "frTitle": "Post d'un projet privé pour admin@test.com",
      "id": "UG9zdDpwb3N0MTY=",
      "title": "Post d'un projet privé pour admin@test.com",
      "translations": Array [
        Object {
          "abstract": null,
          "body": "Very good, approved by la street",
          "locale": "EN_GB",
          "title": "Private project post for admin@test.com",
        },
        Object {
          "abstract": null,
          "body": "Très bien tout ça je valide",
          "locale": "FR_FR",
          "title": "Post d'un projet privé pour admin@test.com",
        },
      ],
    },
    Object {
      "enTitle": "Second post owned by an owner",
      "frTitle": "Deuxieme article possédé par un owner",
      "id": "UG9zdDpwb3N0V2l0aE93bmVyMg==",
      "title": "Deuxieme article possédé par un owner",
      "translations": Array [
        Object {
          "abstract": null,
          "body": "Post body owned by an owner2",
          "locale": "EN_GB",
          "title": "Second post owned by an owner",
        },
        Object {
          "abstract": null,
          "body": "Body de l'article avec owner2",
          "locale": "FR_FR",
          "title": "Deuxieme article possédé par un owner",
        },
      ],
    },
    Object {
      "enTitle": "User's Post on his own proposal",
      "frTitle": "Post d'un utilisateur sur sa proposition",
      "id": "UG9zdDpwb3N0MTc=",
      "title": "Post d'un utilisateur sur sa proposition",
      "translations": Array [
        Object {
          "abstract": null,
          "body": "very good that, I validate",
          "locale": "EN_GB",
          "title": "User's Post on his own proposal",
        },
        Object {
          "abstract": null,
          "body": "Très bien tout ça je valide",
          "locale": "FR_FR",
          "title": "Post d'un utilisateur sur sa proposition",
        },
      ],
    },
    Object {
      "enTitle": "Welcomatic's Post on his own proposal",
      "frTitle": "Post de welcomatic sur sa proposition",
      "id": "UG9zdDpwb3N0MTg=",
      "title": "Post de welcomatic sur sa proposition",
      "translations": Array [
        Object {
          "abstract": null,
          "body": "very good that, I validate",
          "locale": "EN_GB",
          "title": "Welcomatic's Post on his own proposal",
        },
        Object {
          "abstract": null,
          "body": "Très bien tout ça je valide",
          "locale": "FR_FR",
          "title": "Post de welcomatic sur sa proposition",
        },
      ],
    },
    Object {
      "enTitle": "user4's Post on his own proposal",
      "frTitle": "Post de user4 sur sa proposition",
      "id": "UG9zdDpwb3N0MTk=",
      "title": "Post de user4 sur sa proposition",
      "translations": Array [
        Object {
          "abstract": null,
          "body": "very good that, I validate",
          "locale": "EN_GB",
          "title": "user4's Post on his own proposal",
        },
        Object {
          "abstract": null,
          "body": "Très bien tout ça je valide",
          "locale": "FR_FR",
          "title": "Post de user4 sur sa proposition",
        },
      ],
    },
    Object {
      "enTitle": "user5's Post on his own proposal",
      "frTitle": "Remerciment",
      "id": "UG9zdDpwb3N0MjA=",
      "title": "Remerciment",
      "translations": Array [
        Object {
          "abstract": null,
          "body": "Thx all to help us in covid times. This helping me very much !",
          "locale": "EN_GB",
          "title": "user5's Post on his own proposal",
        },
        Object {
          "abstract": null,
          "body": "Merci à tous de nous aider dans des moment pareil. Ça nous aide beaucoup !",
          "locale": "FR_FR",
          "title": "Remerciment",
        },
      ],
    },
    Object {
      "enTitle": "First news",
      "frTitle": "Mon premier article",
      "id": "UG9zdDpwb3N0MjE=",
      "title": "Mon premier article",
      "translations": Array [
        Object {
          "abstract": null,
          "body": "I am a journalist ! Or not.",
          "locale": "EN_GB",
          "title": "First news",
        },
        Object {
          "abstract": null,
          "body": "Je me prends trop pour un journaliste à écrire des articles",
          "locale": "FR_FR",
          "title": "Mon premier article",
        },
      ],
    },
    Object {
      "enTitle": "",
      "frTitle": "",
      "id": "UG9zdDpwb3N0MjI=",
      "title": "",
      "translations": Array [],
    },
    Object {
      "enTitle": "Article créer par un membre du giec",
      "frTitle": "Article créer par un membre du giec",
      "id": "UG9zdDpwb3N0V2l0aE9yZ2FNZW1iZXI=",
      "title": "Article créer par un membre du giec",
      "translations": Array [
        Object {
          "abstract": null,
          "body": "Body de l'article",
          "locale": "FR_FR",
          "title": "Article créer par un membre du giec",
        },
      ],
    },
    Object {
      "enTitle": "Post title owned by an owner",
      "frTitle": "Articles avec owner",
      "id": "UG9zdDpwb3N0V2l0aE93bmVy",
      "title": "Articles avec owner",
      "translations": Array [
        Object {
          "abstract": null,
          "body": "Post body owned by an owner",
          "locale": "EN_GB",
          "title": "Post title owned by an owner",
        },
        Object {
          "abstract": null,
          "body": "Body de l'article avec owner",
          "locale": "FR_FR",
          "title": "Articles avec owner",
        },
      ],
    },
    Object {
      "enTitle": "Actuality of the project théo",
      "frTitle": "Actualité du projet théo",
      "id": "UG9zdDpwb3N0V2l0aE93bmVyT25Qcm9wb3NhbA==",
      "title": "Actualité du projet théo",
      "translations": Array [
        Object {
          "abstract": null,
          "body": "Body of the actuality of the project théo",
          "locale": "EN_GB",
          "title": "Actuality of the project théo",
        },
        Object {
          "abstract": null,
          "body": "Body de l'actualité du projet théo",
          "locale": "FR_FR",
          "title": "Actualité du projet théo",
        },
      ],
    },
    Object {
      "enTitle": "Deuxieme article",
      "frTitle": "Deuxieme article",
      "id": "UG9zdDpwb3N0V2l0aE9yZ2FNZW1iZXIy",
      "title": "Deuxieme article",
      "translations": Array [
        Object {
          "abstract": null,
          "body": "Body de l'article",
          "locale": "FR_FR",
          "title": "Deuxieme article",
        },
      ],
    },
    Object {
      "enTitle": "Post EN 1",
      "frTitle": "Post FR 1",
      "id": "UG9zdDpwb3N0MQ==",
      "title": "Post FR 1",
      "translations": Array [
        Object {
          "abstract": "Abstract of post EN 1",
          "body": "Body of post EN 1",
          "locale": "EN_GB",
          "title": "Post EN 1",
        },
        Object {
          "abstract": "Résumé du post FR 1",
          "body": "Contenu du post FR 1",
          "locale": "FR_FR",
          "title": "Post FR 1",
        },
      ],
    },
    Object {
      "enTitle": "Post EN 10",
      "frTitle": "Post FR 10",
      "id": "UG9zdDpwb3N0MTA=",
      "title": "Post FR 10",
      "translations": Array [
        Object {
          "abstract": "Abstract of post EN 10",
          "body": "Body of post EN 10",
          "locale": "EN_GB",
          "title": "Post EN 10",
        },
        Object {
          "abstract": "Résumé du post FR 10",
          "body": "Contenu du post FR 10",
          "locale": "FR_FR",
          "title": "Post FR 10",
        },
      ],
    },
    Object {
      "enTitle": "Post EN 12",
      "frTitle": "Post FR 12",
      "id": "UG9zdDpwb3N0MTI=",
      "title": "Post FR 12",
      "translations": Array [
        Object {
          "abstract": "Abstract of post EN 12",
          "body": "Body of post EN 12",
          "locale": "EN_GB",
          "title": "Post EN 12",
        },
        Object {
          "abstract": "Résumé du post FR 12",
          "body": "Contenu du post FR 12",
          "locale": "FR_FR",
          "title": "Post FR 12",
        },
      ],
    },
    Object {
      "enTitle": "Post EN 13",
      "frTitle": "Post FR 13",
      "id": "UG9zdDpwb3N0MTM=",
      "title": "Post FR 13",
      "translations": Array [
        Object {
          "abstract": "Abstract of post EN 13",
          "body": "Body of post EN 13",
          "locale": "EN_GB",
          "title": "Post EN 13",
        },
        Object {
          "abstract": "Résumé du post FR 13",
          "body": "Contenu du post FR 13",
          "locale": "FR_FR",
          "title": "Post FR 13",
        },
      ],
    },
    Object {
      "enTitle": "Post EN 14",
      "frTitle": "Post FR 14",
      "id": "UG9zdDpwb3N0MTQ=",
      "title": "Post FR 14",
      "translations": Array [
        Object {
          "abstract": "Abstract of post EN 14",
          "body": "Body of post EN 14",
          "locale": "EN_GB",
          "title": "Post EN 14",
        },
        Object {
          "abstract": "Résumé du post FR 14",
          "body": "Contenu du post FR 14",
          "locale": "FR_FR",
          "title": "Post FR 14",
        },
      ],
    },
    Object {
      "enTitle": "Post EN 15",
      "frTitle": "Post FR 15",
      "id": "UG9zdDpwb3N0MTU=",
      "title": "Post FR 15",
      "translations": Array [
        Object {
          "abstract": "Abstract of post EN 15",
          "body": "Body of post EN 15",
          "locale": "EN_GB",
          "title": "Post EN 15",
        },
        Object {
          "abstract": "Résumé du post FR 15",
          "body": "Contenu du post FR 15",
          "locale": "FR_FR",
          "title": "Post FR 15",
        },
      ],
    },
    Object {
      "enTitle": "Post EN 2",
      "frTitle": "Post FR 2",
      "id": "UG9zdDpwb3N0Mg==",
      "title": "Post FR 2",
      "translations": Array [
        Object {
          "abstract": "Abstract of post EN 2",
          "body": "Body of post EN 2",
          "locale": "EN_GB",
          "title": "Post EN 2",
        },
        Object {
          "abstract": "Résumé du post FR 2",
          "body": "Contenu du post FR 2",
          "locale": "FR_FR",
          "title": "Post FR 2",
        },
      ],
    },
    Object {
      "enTitle": "Post EN 3",
      "frTitle": "Post FR 3",
      "id": "UG9zdDpwb3N0Mw==",
      "title": "Post FR 3",
      "translations": Array [
        Object {
          "abstract": "Abstract of post EN 3",
          "body": "Body of post EN 3",
          "locale": "EN_GB",
          "title": "Post EN 3",
        },
        Object {
          "abstract": "Résumé du post FR 3",
          "body": "Contenu du post FR 3",
          "locale": "FR_FR",
          "title": "Post FR 3",
        },
      ],
    },
    Object {
      "enTitle": "Post EN 4",
      "frTitle": "Post FR 4",
      "id": "UG9zdDpwb3N0NA==",
      "title": "Post FR 4",
      "translations": Array [
        Object {
          "abstract": "Abstract of post EN 4",
          "body": "Body of post EN 4",
          "locale": "EN_GB",
          "title": "Post EN 4",
        },
        Object {
          "abstract": "Résumé du post FR 4",
          "body": "Contenu du post FR 4",
          "locale": "FR_FR",
          "title": "Post FR 4",
        },
      ],
    },
    Object {
      "enTitle": "Post EN 5",
      "frTitle": "Post FR 5",
      "id": "UG9zdDpwb3N0NQ==",
      "title": "Post FR 5",
      "translations": Array [
        Object {
          "abstract": "Abstract of post EN 5",
          "body": "Body of post EN 5",
          "locale": "EN_GB",
          "title": "Post EN 5",
        },
        Object {
          "abstract": "Résumé du post FR 5",
          "body": "Contenu du post FR 5",
          "locale": "FR_FR",
          "title": "Post FR 5",
        },
      ],
    },
    Object {
      "enTitle": "Post EN 6",
      "frTitle": "Post FR 6",
      "id": "UG9zdDpwb3N0Ng==",
      "title": "Post FR 6",
      "translations": Array [
        Object {
          "abstract": "Abstract of post EN 6",
          "body": "Body of post EN 6",
          "locale": "EN_GB",
          "title": "Post EN 6",
        },
        Object {
          "abstract": "Résumé du post FR 6",
          "body": "Contenu du post FR 6",
          "locale": "FR_FR",
          "title": "Post FR 6",
        },
      ],
    },
    Object {
      "enTitle": "Post EN 7",
      "frTitle": "Post FR 7",
      "id": "UG9zdDpwb3N0Nw==",
      "title": "Post FR 7",
      "translations": Array [
        Object {
          "abstract": "Abstract of post EN 7",
          "body": "Body of post EN 7",
          "locale": "EN_GB",
          "title": "Post EN 7",
        },
        Object {
          "abstract": "Résumé du post FR 7",
          "body": "Contenu du post FR 7",
          "locale": "FR_FR",
          "title": "Post FR 7",
        },
      ],
    },
    Object {
      "enTitle": "Post EN 8",
      "frTitle": "Post FR 8",
      "id": "UG9zdDpwb3N0OA==",
      "title": "Post FR 8",
      "translations": Array [
        Object {
          "abstract": "Abstract of post EN 8",
          "body": "Body of post EN 8",
          "locale": "EN_GB",
          "title": "Post EN 8",
        },
        Object {
          "abstract": "Résumé du post FR 8",
          "body": "Contenu du post FR 8",
          "locale": "FR_FR",
          "title": "Post FR 8",
        },
      ],
    },
    Object {
      "enTitle": "Post EN 9",
      "frTitle": "Post FR 9",
      "id": "UG9zdDpwb3N0OQ==",
      "title": "Post FR 9",
      "translations": Array [
        Object {
          "abstract": "Abstract of post EN 9",
          "body": "Body of post EN 9",
          "locale": "EN_GB",
          "title": "Post EN 9",
        },
        Object {
          "abstract": "Résumé du post FR 9",
          "body": "Contenu du post FR 9",
          "locale": "FR_FR",
          "title": "Post FR 9",
        },
      ],
    },
  ],
}
`;
